aspnetcore/host-and-deploy/health-checks.md

```diff
@@ -0,0 +1,359 @@
+
+Register health check services with `AddHealthChecks` in `Startup.ConfigureServices`. Call Health Check Middleware in the app processing pipeline in `Startup.Configure`.
+
+In the sample app, the health check endpoint is created at `/health` (*BasicStartup.cs*):
+
+[!code-csharp[](health-checks/samples/2.x/HealthChecksSample/BasicStartup.cs?name=snippet1&highlight=5,10)]
+
+To run the basic configuration scenario using the sample app, execute the following command from the project's folder in a command shell:
+
+```console
+dotnet run --scenario basic
+```
+
+### Docker example
+
+[Docker](xref:host-and-deploy/docker/index) offers a built-in `HEALTHCHECK` directive that can be used to check the status of an app that uses the basic health check configuration:
+
+```
+HEALTHCHECK CMD curl --fail http://localhost:5000/health || exit
+```
+
+## Create health checks
+
+Health checks are created by implementing the `IHealthCheck` interface. The `IHealthCheck.CheckHealthAsync` method returns a `Task<HealthCheckResult>` of either `Healthy`, `Unhealthy`, `Degraded`, or `Failed`. The result is written as a plain text response with a configurable status code (configuration is described in the [Health check options](#health-check-options) section).
+
+The following `ExampleHealthCheck` class demonstrates the layout of a health check:
+
+```csharp
+public class ExampleHealthCheck : IHealthCheck
+{
+    public ExampleHealthCheck()
+    {
+        // Use DI to supply any required services to the health check.
+    }
+
+    public string Name => "example_check";
+
+    public Task<HealthCheckResult> CheckHealthAsync(
+        CancellationToken cancellationToken = default(CancellationToken))
+    {
+        // Execute health check logic here. This example sets a dummy
+        // variable to true.
+        var healthCheckResultHealthy = true;
+
+        // Return a HealthCheckResult value depending on the outcome of the
+        // check(s). Other HealthCheckResult outcomes not shown in this
+        // example are Degraded and Failed.
+        if (healthCheckResultHealthy)
+        {
+            return Task.FromResult(
+                HealthCheckResult.Healthy("The check indicates a healthy state."));
+        }
+
+        return Task.FromResult(
+            HealthCheckResult.Unhealthy("The check indicates an unhealthy state."));
+    }
+}
+```
+
+Add the `ExampleHealthCheck` with `AddCheck` and call `UseHealthChecks` in the processing pipeline with the endpoint URL:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHealthChecks()
+        .AddCheck(new ExampleHealthCheck);
```

Comment:
This example should be different - it should be `AddCheck<ExampleHealthCheck>()` because it metnions DI. If you construct the health check and pass it it the build it won't get DI services.

Reply:
I'd suggest looking over these again, as they changed significantly https://github.com/aspnet/Diagnostics/blob/release/2.2/src/Microsoft.AspNetCore.Diagnostics.HealthChecks/Builder/HealthCheckApplicationBuilderExtensions.cs

Reply:
> as they changed significantly

Sure thing ... I was following Glen's lead based on the blog post, which is probably *old* given the latest bits.